### PR TITLE
remove www from domain before checking it

### PR DESCRIPTION
**Pull request by gmethvin**

### CODE REVIEW

File: shoebox/app/com/keepit/classify/DomainClassifier.scala [hunk 1]
```diff
@@ -89,7 +89,13 @@ class DomainClassifierImpl @Inject()(system: ActorSystem, db: Database, client:
     new DomainClassificationActor(db, client, updater, domainRepo, tagRepo, domainToTagRepo)
   })
 
-  def isSensitive(domainName: String): Either[Future[Boolean], Boolean] = {
+  private val splitPattern = """\.""".r
+
+  def isSensitive(hostname: String): Either[Future[Boolean], Boolean] = {
+    val domainName = (splitPattern.split(hostname.toLowerCase).toSeq match {
+      case "www" +: domain => domain
```
Comment (eishay) [inline shoebox/app/com/keepit/classify/DomainClassifier.scala:96]: nice

**Comment by 2is10** [comment]
Generalizing from this pull request suggests that, to determine the sensitivity of a hostname, perhaps we should be checking the sensitivity of the hostname _and_ its higher-level domains, and declaring it sensitive if any of them are sensitive.

For example:
  www.myblog.blogger.com → www.myblog.blogger.com, myblog.blogger.com, blogger.com, com
  someone.somenewsite.xxx → someone.somenewsite.xxx, somenewsite.xxx, xxx

**Comment by gmethvin** [comment]
@2is10 Based on what Barak told me, www is a special case. Any other subdomains should be searched as the whole hostname.

**Comment by eishay** [comment]
Agree with both of you, we should ask for domains explicitly (with the www
exception).
If the domain is not there and we want to return a fast result, not waiting
for the third party api, then the higher level domain sounds like a
good heuristics.
What do you think Greg?

**Pull request merged**